dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -102,9 +107,21 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
     {
         GetToastHandlers().remove(token);
     }
-    void ToastNotificationManager::ShowToast(winrt::Microsoft::Windows::ToastNotifications::ToastNotification const& /* toast */)
+    void ToastNotificationManager::ShowToast(winrt::Microsoft::Windows::ToastNotifications::ToastNotification const& toast)
     {
-        throw hresult_not_implemented();
+        THROW_HR_IF(WPN_E_NOTIFICATION_POSTED, toast.ToastId() != 0);
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::INotificationProperties> notificationProperties
+            = winrt::make_self<NotificationProperties>(toast);
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::INotificationTransientProperties> notificationTransientProperties
+            = winrt::make_self<NotificationTransientProperties>(toast);
+
+        auto toastAppId{ RetrieveAppId() };
+
+        DWORD notificationId = 0;
+        THROW_IF_FAILED(ToastNotifications_PostToast(toastAppId.c_str(), notificationProperties.get(), notificationTransientProperties.get(), &notificationId));
+        toast.ToastId(notificationId);
```

Comment:
Can we do a quick check here since this is critical code: THROW_HR_IF(E_UNEXPECTED, notificationId == 0);

Reply:
Good catch!